sdk/identity/Azure.Identity/src/AadIdentityClient.cs

```diff
@@ -42,68 +44,87 @@ public virtual async Task<AccessToken> AuthenticateAsync(string tenantId, string
                 }
-
-                throw response.CreateRequestFailedException();
             }
         }
 
         public virtual async Task<AccessToken> AuthenticateAsync(string tenantId, string clientId, X509Certificate2 clientCertificate, string[] scopes, CancellationToken cancellationToken = default)
         {
             using (Request request = CreateClientCertificateAuthRequest(tenantId, clientId, clientCertificate, scopes))
             {
-                var response = await _pipeline.SendRequestAsync(request, cancellationToken).ConfigureAwait(false);
-
-                if (response.Status == 200 || response.Status == 201)
+                try
                 {
-                    var result = await DeserializeAsync(response.ContentStream, cancellationToken).ConfigureAwait(false);
-
-                    return new Response<AccessToken>(response, result);
+                    return await SendAuthRequestAsync(request, cancellationToken).ConfigureAwait(false);
+                }
+                catch (RequestFailedException ex)
+                {
+                    throw new AuthenticationFailedException(AuthenticationRequestFailedError, ex);
                 }
-
-                throw await response.CreateRequestFailedExceptionAsync();
             }
         }
 
         public virtual AccessToken Authenticate(string tenantId, string clientId, X509Certificate2 clientCertificate, string[] scopes, CancellationToken cancellationToken = default)
         {
             using (Request request = CreateClientCertificateAuthRequest(tenantId, clientId, clientCertificate, scopes))
             {
-                var response = _pipeline.SendRequest(request, cancellationToken);
-
-                if (response.Status == 200 || response.Status == 201)
+                try
                 {
-                    var result = Deserialize(response.ContentStream);
-
-                    return new Response<AccessToken>(response, result);
+                    return SendAuthRequest(request, cancellationToken);
+                }
+                catch (RequestFailedException ex)
+                {
+                    throw new AuthenticationFailedException(AuthenticationRequestFailedError, ex);
                 }
+            }
+        }
+        private async Task<AccessToken> SendAuthRequestAsync(Request request, CancellationToken cancellationToken)
+        {
+            var response = await _pipeline.SendRequestAsync(request, cancellationToken).ConfigureAwait(false);
+
+            if (response.Status == 200 || response.Status == 201)
+            {
+                var result = await DeserializeAsync(response.ContentStream, cancellationToken).ConfigureAwait(false);
 
-                throw response.CreateRequestFailedException();
+                return new Response<AccessToken>(response, result);
             }
+
+            throw await response.CreateRequestFailedExceptionAsync();
```

Comment:
Does this require a .ConfigureAwait(false) (and everywhere we call something like that?)